app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepostoryImpl.kt

```diff
@@ -50,27 +52,43 @@ class AgendaRepositoryImpl @Inject constructor(
         if (deletedSuccessfully) {
             return eventRepository.deleteFinallyEventIds(deletedEventIds)
         } else {
-            return RepositoryResult.Error("Failed to sync agenda - deleteFinallyEventIds")
+            return RepositoryResult.Error(UiText.ResOrStr(R.string.agenda_sync_error, "Failed to sync agenda - deleteFinallyEventIds"))
         }
     }
 
-    override suspend fun createEvent(event: AgendaItem.Event): RepositoryResult {
+    override suspend fun createEvent(event: AgendaItem.Event): RepositoryResult<AgendaItem.Event> {
         return eventRepository.createEvent(event)
     }
 
     override suspend fun getEvent(eventId: EventId): AgendaItem.Event? {
         return eventRepository.getEvent(eventId)
     }
 
-    override suspend fun updateEvent(event: AgendaItem.Event): RepositoryResult {
+    override suspend fun updateEvent(event: AgendaItem.Event): RepositoryResult<AgendaItem.Event> {
         return eventRepository.updateEvent(event)
     }
 
-    override suspend fun deleteEventId(eventId: EventId): RepositoryResult {
+    override suspend fun deleteEventId(eventId: EventId): RepositoryResult<AgendaItem.Event> {
         return eventRepository.deleteEventId(eventId)
     }
 
-    override suspend fun clearAllEvents(): RepositoryResult {
+    override suspend fun clearAllEvents(): RepositoryResult<Void> {
         return eventRepository.clearAllEvents()
     }
+
+    override suspend fun confirmAttendeeExists(
+        attendeeEmail: Email,
+        onSuccess: (attendee: Attendee?) -> Unit,
+        onFailure: (error: UiText) -> Unit
+    ) {
+        val result = attendeeRepository.getAttendee(attendeeEmail)
+        when(result) {
+            is RepositoryResult.Success -> {
+                onSuccess(result.data)
+            }
+            is RepositoryResult.Error -> {
+                onFailure(result.message)
+            }
+        }
```

Comment:
Regarding your question: Sure, this works and you can do it like this. But then I wonder why you use `RepositoryResult` at all, since I don't see a reason to only use callbacks for this single function (if there is from your side, please correct me)

Reply:
Good point, I will change to use the Result instead and do the checks on the ViewModel side. I think I got bad thinking habits from using callbacks for so long!